### PR TITLE
Add Game Networking

### PR DESCRIPTION
**Pull request by MFatihMAR**
https://github.com/MFatihMAR/Awesome-Game-Networking

Game networking is a subset of computer networking that covers data replication, state synchronization, snapshot interpolation, client prediction, lag compensation, bandwidth optimization and many more techniques involved in multiplayer online gameplay programming.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -336,6 +336,7 @@
 ## Gaming
 
 - [Game Development](https://github.com/ellisonleao/magictools#readme)
+- [Game Networking](https://github.com/MFatihMAR/Awesome-Game-Networking#readme) - Network programming for multiplayer online games.
```
Comment (MFatihMAR) [inline readme.md]: I may or may not remove the description if you think it's a little bit long.

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/MFatihMAR/Awesome-Game-Networking
✖ Linting

  README.md:75:3
  ✖    9:1  First heading should be name of awesome list                       remark-lint:awesome/toc
  ✖   75:3  Link to https://www.udemy.com/unrealengine-cpp/ is dead            remark-lint:no-dead-urls
  ✖   76:3  Link to https://www.udemy.com/unrealmultiplayer/ is dead           remark-lint:no-dead-urls
  ✖  114:3  Link to https://wiki.linuxfoundation.org/networking/netem is dead  remark-lint:no-dead-urls

  4 errors
```

https://github.com/sindresorhus/awesome-lint

File: readme.md (hunk 1 above)
Comment (sindresorhus) [reply readme.md]: It's fine to keep it.

**Comment by sindresorhus** [comment]
Instead of `C# / Lidgren.Network`, I would make it a sub-list:


- C#
	- Lidgren.Network

**Comment by sindresorhus** [comment]
Otherwise, looks good :)

**Comment by MFatihMAR** [comment]
> First heading should be name of awesome list

Well, it has full-width SVG banner for the logo (and name) instead.

> Link to https://www.udemy.com/unrealengine-cpp/ is dead

This one is interesting because they are not dead, I manually checked links and also  double-checked `README.md` but getting these errors. (I'm confused, they are alive)

> Instead of `C# / Lidgren.Network`, I would make it a sub-list

Sure, good idea. I will update the whole `Libraries` section ASAP - thanks @sindresorhus !

**Comment by MFatihMAR** [comment]
I updated all sections and they have **cool** subsections now.

Thanks for the feedback @sindresorhus :)

**Comment by sindresorhus** [comment]
> This one is interesting because they are not dead, I manually checked links and also double-checked `README.md` but still getting these errors. (I'm confused, they are alive!)

https://github.com/transitive-bullshit/check-links/issues/4

**Comment by sindresorhus** [comment]
> Well, it has full-width SVG banner for the logo (and name) instead.

Right. Ignore that.

**Comment by sindresorhus** [comment]
Looks good :)

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1100459327536545792

**Pull request merged**